purchase_order_product_recommendation_brand/wizards/purchase_order_recommendation.py

```diff
@@ -0,0 +1,47 @@
+# Copyright 2019 Tecnativa
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from odoo import api, fields, models
+
+
+class PurchaseOrderRecommendation(models.TransientModel):
+    _inherit = 'purchase.order.recommendation'
+
+    product_brand_ids = fields.Many2many(
+        comodel_name='product.brand',
+        string="Brands",
+    )
+
+    def _get_products(self):
+        """Filter products of the given brands"""
+        products = super()._get_products()
+        # Filter products by brand if set.
+        # It will apply to show_all_partner_products as well
+        if self.product_brand_ids:
```

Comment:
Why do you need this if you are already filtering in the domain?

Reply:
In this method , I filter the supplier products (given by the proper supplierinfos). In `_get_all_products_domain` we obtain all the products independently of the supplier.

Reply:
OK, I'm not sure about the performance of this.